src/services/expense.service.js

```diff
@@ -0,0 +1,80 @@
+const { Expense } = require('../models/Expense.model');
+const { Op } = require('sequelize');
+
+const getAll = async (category, userId, from, to) => {
+  const whereClause = {};
+
+  if (category) {
+    whereClause.category = category;
+  }
+
+  if (userId) {
+    whereClause.userId = userId;
+  }
+
+  if (from || to) {
+    whereClause.spentAt = {};
+
+    if (from) {
+      whereClause.spentAt[Op.gte] = new Date(from);
+    }
+
+    if (to) {
+      whereClause.spentAt[Op.lte] = new Date(to);
+    }
+  }
+
+  const result = await Expense.findAll({
+    where: whereClause,
+  });
+
+  return result;
+};
+
+const getById = async (id) => {
+  const result = await Expense.findByPk(id);
+
+  return result;
+};
+
+const createExpense = async ({
+  userId,
+  title,
+  amount,
+  category,
+  note,
+  spentAt,
+}) => {
```

Comment:
Consider adding validation for the input data in the `createExpense` function to ensure all required fields are present and valid before creating a new expense.